src/main/java/gyro/aws/apigatewayv2/DomainNameResource.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2020, Brightspot.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.aws.apigatewayv2;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import gyro.aws.AwsCredentials;
+import gyro.aws.AwsResource;
+import gyro.aws.Copyable;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Id;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Required;
+import software.amazon.awssdk.services.acm.AcmClient;
+import software.amazon.awssdk.services.apigatewayv2.ApiGatewayV2Client;
+import software.amazon.awssdk.services.apigatewayv2.model.DomainName;
+import software.amazon.awssdk.services.apigatewayv2.model.GetDomainNamesResponse;
+
+/**
+ * Create a domain name.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     aws::domain-name example-domain-name
+ *         name: "vpn.ops-test.psdops.com"
+ *
+ *         domain-name-configurations
+ *             certificate: 'arn:aws:acm:us-east-2:572681481110:certificate/c7de099f-599d-4112-b813-e731fdef0760'
+ *             endpoint-type: REGIONAL
+ *         end
+ *
+ *         tags: {
+ *             "example-key-change": "example-value"
+ *         }
+ *     end
+ */
+@Type("domain-name")
```

Comment:
Simiarr as before to make it less generic
```suggestion
@Type("api-gateway-domain-name")
```